stable_baselines3/common/vec_env/vec_normalize.py

```diff
@@ -159,7 +160,11 @@ def set_venv(self, venv: VecEnv) -> None:
         """
         if self.venv is not None:
             raise ValueError("Trying to set venv of already initialized VecNormalize wrapper.")
-        VecEnvWrapper.__init__(self, venv)
+        self.venv = venv
+        self.num_envs = venv.num_envs
+        venv.observation_space = self.observation_space
```

Comment:
is that needed, if we don't overwrite `self.observation_space`, we should be fine

Reply:
It fails here otherwise.

https://github.com/DLR-RM/stable-baselines3/blob/d82b3d566273cee08f840dd3ead9eabe8efd17a6/stable_baselines3/common/vec_env/vec_normalize.py#L169-L170

Reply:
> I see, it's just that modifying the underlying obs space doesn't sounds like a good side effect (once we do that, there is no way to know what was the original obs space).

Agree

For what it's worth, all tests pass when venv spaces aren't modified and check skipped

Reply:
> do you think there is an easy way to know when we modify the obs space? (so we can skip/relax the check instead of overriding the venv obs space)

I'll think about it. Maybe a warning could suit:

```txt
the loaded model is normalized and the observation space does not match the set model. We strongly recommend to use this new send instead:
venv = model.get_env()
```
But this is not the ideal solution, because there would be no way to "solve" the warning for the user.

Reply:
On second thought, I think it's just the check that has no reason to be. There is no reason for the wrapped observation space to be the same as the unwrapped. I would simply remove those lines and put no warning.

Reply:
for me, only the shape check makes sense